modulus/utils/filesystem.py

```diff
@@ -43,11 +46,88 @@ def _get_fs(path):
         return fsspec.filesystem("file")
 
 
-def _download_cached(path: str, recursive: bool = False) -> str:
+def _download_ngc_model_file(path: str, out_path: str, timeout: int = 300) -> str:
+    """Pulls files from model registry on NGC. Supports private registries when NGC
+    API key is set the the API_KEY enviroment variable. If download file is a zip folder
+    it will get unzipped.
+
+    Args:
+        path (str): NGC model file path of form `ngc://org/team/model/version/filename`
```

Comment:
This is nice! An enhancement might be to align this string format with the huggingface filesystem abstraction so it is more familiar to users.
https://huggingface.co/docs/huggingface_hub/main/en/guides/hf_file_system

Reply:
Yeah, I agree this would be really nice to do. Trouble is the NGC org -> team concept 

Also seems HF isnt fully consistent between datasets and models. I like the idea of trying to follow:
```
hf://[<repo_type_prefix>]<repo_id>[@<revision>]/<path/in/repo>
```
(repo_id is essentially org/team/name of registry)


So I'm thinking:

<repo_id> = `org/team/name of repo`
@<revision> = version
<repo_type_prefix> = 'models' or 'datasets'

Reply:
Yeah, I like that. `<repo_type_prefix>` should probably have a `/` at the end though.

So in the test here, we would have something like this?
```
ngc://models/nvidia/modulus/modulus_dlwp_cubesphere@v0.2/dlwp_cubesphere.zip
```

Reply:
Implemented, clean up the valid url check too with some chatgpt regex and added unit test to check invalid urls.